### ISSUE TITLE
Unable to leave guilds

### ISSUE THREAD

**Issue by WezCorp** [HabitRPG/habitica#5644]
User: 6aaf4e6d-0395-49fb-bd6b-641a82b73df9

Unable to leave guilds - click on the leave button and the pop-up box which showed either confirm or cancel doesn't pop up, and the webpage just goes through it's reload type phase when clicking a task complete for example.
Currently using an up to date version of chrome - but getting loads of java errors. I'll submit a new request for being unable to filter challenges, but looking at some of the java errors it might be related (but I unfortunately don't know enough about java). See below java error which might be related.

TypeError: Cannot read property 'habitrpg' of undefined
    at n (https://habitrpg.com/app-d0549504.js:32:20332)
    at b.filterChallenges (https://habitrpg.com/app-d0549504.js:32:19819)
    at Array.filter (native)
    at https://habitrpg.com/app-d0549504.js:7:26882
    at g (https://habitrpg.com/app-d0549504.js:7:2023)
    at Object.f (https://habitrpg.com/app-d0549504.js:7:9077)
    at j.$digest (https://habitrpg.com/app-d0549504.js:7:15836)
    at j.$apply (https://habitrpg.com/app-d0549504.js:7:17528)
    at $$debounceViewValueCommit (https://habitrpg.com/app-d0549504.js:8:19143)
    at $setViewValue (https://habitrpg.com/app-d0549504.js:8:18829)
##

**Comment by Alys** [comment]
For our reference, this is the filter challenges issue: https://github.com/HabitRPG/habitrpg/issues/5645

@WezCorp Can you please log in to HabitRPG from another browser (preferably one you've never used HabitRPG in before) and see if the error still happens?

**Comment by Alys** [comment]
If that doesn't help as a temporary measure, please show us a screenshot of your guilds page so that we can see what guilds you're in.

Note to devs: he's not a joined member of the `habitrpg` Tavern guild.

**Comment by crookedneighbor** [comment]
@Alys since the error is habitrpg as a property, my guess is it's [related to chat notifications for the tavern](https://github.com/HabitRPG/habitrpg/blob/develop/website/public/js/services/chatServices.js#L26) (since it's id is habitrpg). Though, I did a look up for the guilds that @WezCorp is in and did not see the tavern as one of them.

**Comment by crookedneighbor** [comment]
Beat me to it.

**Comment by WezCorp** [comment]
So I tried using Safari and couldn't even log on - but I've used Safari before. Chrome mobile also had the same error. I downloaded Firefox and still the same errors there.
Should I join the tavern guild?
Also here's the guilds I am currently in - 
The Scholars
Life Hackers
Challenge... Accepted?
The Health Nuts
The Overachievers
Duolinguists of Habitica
📜 Library of Shared Lists (Currently being overhaul by the Ministry of Magic)
The Nerd Fitness Rebellion (Unofficial)Short-Term Goal AccountabilityFighters GuildWeight Loss/Maintenance
Parlons Français!
SuperBetter
Video Producer's Organization
Learning French - Apprendre le français
WezCorp's Guild

**Comment by Alys** [comment]
Thanks for the details!

Don't join the Tavern guild. I mentioned it as one possible source of a problem.

**Comment by WezCorp** [comment]
I found the problem out a few weeks ago when I tried to leave the Fighter's Guild (which I forgot to put on it's own line above, and only joined to find a challenge that was stuck in my 'done' to do list - separate issue which I did a work around for).
I thought it was just because I hadn't updated my chrome, and didn't worry about it for a while.
Also the bottom 2 guild (Learning French and WezCorp's Guild) are both mine - in case that has anything to do with it.

**Comment by Alys** [comment]
@WezCorp Could you please go to the [challenges page](https://habitrpg.com/#/options/groups/challenges), and on the left-hand side, ensure that all of the groups are ticked (they should be by default), then select "participating" in the Membership section. Please then take as many screenshots as necessary to show us all of the challenge titles (no need to show us any other details of the challenges), and drag the screenshots into a comment here. Don't attach them to an email because GitHub ignores email attachments.

**Comment by Alys** [comment]
for our reference: WezCorp's challenges object contains 23 challenges that no longer exist. I'm wondering if that's causing the problem. I'll clear them out after I've compared what I'm seeing in the database with the screenshots.

**Comment by WezCorp** [comment]
So in the challenges screen, there's no place I can actually tick - the guild filter isn't there at all to select which guild's challenges I want to see or not.
I think maybe it might have something to do with the 23 challenges thing - the first time I checked the java errors there were 23. Coincidence?

**Comment by WezCorp** [comment]
![2015-07-20 06 20 23 1](https://cloud.githubusercontent.com/assets/13398985/8767699/7416e162-2ea8-11e5-9ee0-0720e0def3b4.jpg)

**Comment by WezCorp** [comment]
It could be because of my guild - WezCorps guild. I created it to create 100 challenges for myself over the year, but because of the todo list known bug it was throwing stuff around all the time, so I left them all and consolidated them in my to do list - not as challenges

**Comment by Alys** [comment]
@WezCorp Can you please take screenshots of the whole [challenges page](https://habitrpg.com/#/options/groups/challenges)? I'd like to see everything on it in context. First please go to Settings -> Site and untick "show header" so that the screenshots can capture more of the page's content.

Please also take a screenshot of the guilds page. This doesn't need to be the whole list of guilds, just the top part of the page - what you see when in one screen when you've scrolled to the top.

**Comment by WezCorp** [comment]
![picture device independent bitmap 1](https://cloud.githubusercontent.com/assets/13398985/8768185/51de4cf4-2eba-11e5-9580-836df0c17974.jpg)
So unfortunately my workplace uses the oldest version of IE available so it has always shown a bit funny - but this is similar to what I see in chrome anyways

**Comment by WezCorp** [comment]
![picture device independent bitmap 1](https://cloud.githubusercontent.com/assets/13398985/8768474/22dd5fd4-2ec5-11e5-82bd-bfa9df1f1bab.jpg)

**Comment by Alys** [comment]
@WezCorp Can you please take screenshots of the _whole_ challenges page? I'd like to see everything on it in context. Please use Chrome so that the IE bug doesn't interfere, and untick "show header" again so that fewer screenshots are needed.

**Comment by WezCorp** [comment]
![screen shot 2015-07-26 at 8 24 05 am](https://cloud.githubusercontent.com/assets/13398985/8891684/6037ddc0-3378-11e5-8786-0f00aae516c8.png)
[removed now that I've seen them -- Alys]
![screen shot 2015-07-26 at 9 25 38 am](https://cloud.githubusercontent.com/assets/13398985/8891749/6225b24c-3378-11e5-9f56-dba37da566c3.png)

**Comment by WezCorp** [comment]
Would it be easier to kick/remove me from ALL challenges and I just rejoin the ones that I want?

**Comment by eburcat** [comment]
+1
I'm also running into this problem, subscribed to different guilds

**Comment by eburcat** [comment]
Ha, eventually I managed to leave guilds by using the iOS app.

**Comment by WezCorp** [comment]
Same, well I managed to leave most guilds except the private one I made myself - still trapped in that one. I actually think that one is the source of the problem. And it means I still can't filter challenges etc

**Comment by Alys** [comment]
@WezCorp Sorry! I've removed you from that private guild. Has that helped with the challenges?

**Comment by WezCorp** [comment]
@Alys Unfortunately not - I'm not sure what's causing it. Is it easily possible to boot me from all challenges I'm currently in?

**Comment by Alys** [comment]
@WezCorp I've removed you from challenges. You still have the challenge tasks though - let me know if you need help cleaning them up.

**Comment by WezCorp** [comment]
@Alys Unfortunately I still seem to have the same problem. _sigh_
I can live with it though, just means manually searching through challenges sometimes haha

**Comment by Alys** [comment]
So you're still having this error? "Unable to filter challenges - it doesn't show any of the guilds I'm in, so the filter doesn't work at all, can't clicked owned or participating and have it change."

From what I can tell, you're in these groups:
- [your party](https://habitica.com/#/options/groups/guilds/42599d3d-3058-44c3-bc22-07d007499d5f)
- [Learning French - Apprendre le fran√ßais](https://habitica.com/#/options/groups/guilds/3272dfab-2b42-4ccd-ab12-eeeb4aed63ad)
- [Long-Term Goal Accountability](https://habitica.com/#/options/groups/guilds/7c6fcc88-9ff0-47d1-9fca-80e3091c225b)
- [The Language Hackers](https://habitica.com/#/options/groups/guilds/283171a5-422c-4991-bc78-95b1b5b51629)
- [Short-Term Goal Accountability](https://habitica.com/#/options/groups/guilds/59f699a3-9d22-4d6d-b34a-8fcdf2ad1dac)

And you own these challenges:
- [Legendary Mode](https://habitica.com/#/options/groups/challenges/90444514-f587-4294-8070-0cb6a54d730a) - in your party
- [Walk to Mordor](https://habitica.com/#/options/groups/challenges/20031cf0-6a6d-4baf-9e50-a624358ed673) - in your party
- [Learning French](https://habitica.com/#/options/groups/challenges/a031d2d0-20f8-4c8f-90f8-37318ce37148) - in Learning French - Apprendre le fran√ßais
- [Runkeeper Beginner 5K](https://habitica.com/#/options/groups/challenges/2f1e7c4e-af0f-4c01-b909-6dae25ac2529) - this was not in any group but I have moved it to your party
- [One does not simply walk to Mordor](https://habitica.com/#/options/groups/challenges/72af0c7f-794a-4c41-9fa1-472c8e2285ad) - this was not in any group but I have moved it to your party

The final two challenges weren't associated with any group, so I've moved them to your party. Maybe that's what was causing the problem. Is there any improvement in challenge filtering now?

**Comment by Alys** [comment]
@WezCorp Did the changes I made to those challenges fix the problem?

**Comment by WezCorp** [comment]
@Alys Sorry about the delayed response, didn't have time to test until now. But it's BRILLIANT! Worked a charm. I was able to filter challenges in guilds and participating. I deleted those 2 challenges that were causing problems and it still worked. I was also able to join and leave a guild (I couldn't leave them before). So it seems like everything is fixed and I won't avoid challenges as much now haha.
